From bed5c5f1f4c2898696b9f9ab9c05a078d1ef0744 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 13:49:12 +0800
Subject: [PATCH 274/316] AOSCOS: staging: sb105x: add missing function
 prototypes

A few functions in drivers/staging/sb105x/sb_ser_core.h were missing
function prototype declarations. Mark them as static to suppress a couple
of `-Werror=missing-prototype' errors.

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_ser_core.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/sb105x/sb_ser_core.h b/drivers/staging/sb105x/sb_ser_core.h
index c8fb99173299..8835415e1d3d 100644
--- a/drivers/staging/sb105x/sb_ser_core.h
+++ b/drivers/staging/sb105x/sb_ser_core.h
@@ -196,13 +196,13 @@ struct uart_driver {
         struct tty_driver               *tty_driver;
 };
 
-void sb_uart_write_wakeup(struct sb_uart_port *port)
+static void sb_uart_write_wakeup(struct sb_uart_port *port)
 {
     struct sb_uart_info *info = port->info;
     tasklet_schedule(&info->tlet);
 }
 
-void sb_uart_update_timeout(struct sb_uart_port *port, unsigned int cflag,
+static void sb_uart_update_timeout(struct sb_uart_port *port, unsigned int cflag,
 			 unsigned int baud)
 {
     unsigned int bits;
@@ -240,7 +240,7 @@ void sb_uart_update_timeout(struct sb_uart_port *port, unsigned int cflag,
 
     port->timeout = (HZ * bits) / baud + HZ/50;
 }
-unsigned int sb_uart_get_baud_rate(struct sb_uart_port *port, struct MP_TERMIOS *termios,
+static unsigned int sb_uart_get_baud_rate(struct sb_uart_port *port, struct MP_TERMIOS *termios,
 				struct MP_TERMIOS *old, unsigned int min,
 				unsigned int max)
 {
@@ -304,7 +304,7 @@ unsigned int sb_uart_get_baud_rate(struct sb_uart_port *port, struct MP_TERMIOS
 
         return 0;
 }
-unsigned int sb_uart_get_divisor(struct sb_uart_port *port, unsigned int baud)
+static unsigned int sb_uart_get_divisor(struct sb_uart_port *port, unsigned int baud)
 {
         unsigned int quot;
 
-- 
2.48.1

